default/python/AI/InvasionAI.py

```diff
@@ -360,14 +360,37 @@ def evaluate_invasion_planet(planet_id, empire, secure_fleet_missions, verbose=T
             supply_val *= 0.2
 
     threat_factor = min(1, 0.2*MilitaryAI.totMilRating/(sys_total_threat+0.001))**2  # devalue invasions that would require too much military force
-    build_time = 4
 
-    planned_troops = troops if system_secured else min(troops+max_jumps+build_time, max_troops)
-    if not tech_is_complete("SHP_ORG_HULL"):
-        troop_cost = math.ceil(planned_troops/6.0) * (40*(1+foAI.foAIstate.shipCount * AIDependencies.SHIP_UPKEEP))
+    design_id, _, locs = ProductionAI.get_best_ship_info(PriorityType.PRODUCTION_INVASION)
+    if not locs or not universe.getPlanet(locs[0]):
+        # We are in trouble anyway, so just calculate whatever approximation...
+        build_time = 4
+        planned_troops = troops if system_secured else min(troops + max_jumps + build_time, max_troops)
+        if not tech_is_complete("SHP_ORG_HULL"):
+            troop_cost = math.ceil(planned_troops / 6.0) * 40 * FleetUtilsAI.get_fleet_upkeep()
+        else:
+            troop_cost = math.ceil(planned_troops / 6.0) * 20 * FleetUtilsAI.get_fleet_upkeep()
+    else:
+        loc = locs[0]
+        species_here = universe.getPlanet(loc).speciesName
+        design = fo.getShipDesign(design_id)
+        cost_per_ship = design.productionCost(empire_id, loc)
+        build_time = design.productionTime(empire_id, loc)
+        troops_per_ship = CombatRatingsAI.weight_attack_troops(design.troopCapacity,
+                                                               CombatRatingsAI.get_species_troops_grade(species_here))
+        planned_troops = troops if system_secured else min(troops + max_jumps + build_time, max_troops)
+        ships_needed = math.ceil(planned_troops / float(troops_per_ship))
+        troop_cost = ships_needed * cost_per_ship  # fleet upkeep is already included in query from server
+
+    # apply some negative bias to expensive operations
+    normalized_cost = float(troop_cost) / max(empire.productionPoints, 1)
+    if normalized_cost < 1:
+        cost_score = 0
     else:
-        troop_cost = math.ceil(planned_troops/6.0) * (20*(1+foAI.foAIstate.shipCount * AIDependencies.SHIP_UPKEEP))
-    planet_score = retaliation_risk_factor(planet.owner) * threat_factor * max(0, pop_val+supply_val+bld_tally+enemy_val-0.8*troop_cost)
+        cost_score = -(normalized_cost**2 / 50.0) * troop_cost
+
+    base_score = max(0, pop_val + supply_val + bld_tally + enemy_val - cost_score)
```

Comment:
Define "underweight". 

The basic idea is that producing troops with a total cost in the order of the Empire's production output does not cause much struggle:
Assume that troop ships build in 3 turns, then if the cost is less than 3 turns of empire's production output, there will remain (some) PP for other projects.
If the cost is even within a single turn's production output, then it will take less than a third of our resources per turn. It is in some sense negliglible and it would be preferable to get the best planet at that point instead of one where we have to spend 20% less PP.

 If it is slightly higher cost per turn than we can produce, we pause the production of other things for some 3-5 turns which does not hurt too much and the opportunity cost is thus limited.
The longer we have to build troop ships because we do not have "sufficient" PP, the later we will get returns and the higher will be the opportunity cost.

An extreme case was highlighted by Mat with regard to the Ancient Guardians:
https://github.com/freeorion/freeorion/issues/1012#issuecomment-279231433

It thus makes sense to scale the ```cost_score``` with the available production.
The numeric factor of 1/50 is arbitrary in the sense that I chose some number that seemed reasonable and led to seemingly decent results ingame.
The quadratic scaling is based on the idea that the opportunity costs seem to warrant a higher-than-linear scaling while not scaling too aggressive.

If we can agree on the general concept, the numbers can be tweaked as needed.